tests/CMake/SiloMakeCheckRunner.cmake

```diff
@@ -0,0 +1,202 @@
+# modification, are permitted provided that the following conditions
+# are met:
+# 
+#    * Redistributions of source code must retain the above copyright
+#      notice, this list of conditions and the disclaimer below.
+#    * Redistributions in binary form must reproduce the above copyright
+#      notice, this list of conditions and the disclaimer (as noted
+#      below) in the documentation and/or other materials provided with
+#      the distribution.
+#    * Neither the name of the LLNS/LLNL nor the names of its
+#      contributors may be used to endorse or promote products derived
+#      from this software without specific prior written permission.
+# 
+# THIS SOFTWARE  IS PROVIDED BY  THE COPYRIGHT HOLDERS  AND CONTRIBUTORS
+# "AS  IS" AND  ANY EXPRESS  OR IMPLIED  WARRANTIES, INCLUDING,  BUT NOT
+# LIMITED TO, THE IMPLIED  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A  PARTICULAR  PURPOSE ARE  DISCLAIMED.  IN  NO  EVENT SHALL  LAWRENCE
+# LIVERMORE  NATIONAL SECURITY, LLC,  THE U.S.  DEPARTMENT OF  ENERGY OR
+# CONTRIBUTORS BE LIABLE FOR  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR  CONSEQUENTIAL DAMAGES  (INCLUDING, BUT NOT  LIMITED TO,
+# PROCUREMENT OF  SUBSTITUTE GOODS  OR SERVICES; LOSS  OF USE,  DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER  IN CONTRACT, STRICT LIABILITY,  OR TORT (INCLUDING
+# NEGLIGENCE OR  OTHERWISE) ARISING IN  ANY WAY OUT  OF THE USE  OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# 
+# This work was produced at Lawrence Livermore National Laboratory under
+# Contract No.  DE-AC52-07NA27344 with the DOE.
+# 
+# Neither the  United States Government nor  Lawrence Livermore National
+# Security, LLC nor any of  their employees, makes any warranty, express
+# or  implied,  or  assumes  any  liability or  responsibility  for  the
+# accuracy, completeness,  or usefulness of  any information, apparatus,
+# product, or  process disclosed, or  represents that its use  would not
+# infringe privately-owned rights.
+# 
+# Any reference herein to  any specific commercial products, process, or
+# services by trade name,  trademark, manufacturer or otherwise does not
+# necessarily  constitute or imply  its endorsement,  recommendation, or
+# favoring  by  the  United  States  Government  or  Lawrence  Livermore
+# National Security,  LLC. The views  and opinions of  authors expressed
+# herein do not necessarily state  or reflect those of the United States
+# Government or Lawrence Livermore National Security, LLC, and shall not
+# be used for advertising or product endorsement purposes.
+#
+##############################################################################
+
+cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
+
+# Executes a test and prints failure/success.
+# May need more args as other tests (fortran, python, special, json) are added
+function(silo_add_make_check_runner)
```

Comment:
Is this the top entry point that replaces the AutoTools AutoTest stuff with CMake's testing stuff?

Reply:
I wouldn't say it is using 'CMake's testing stuff'.  That would really be using CTest, which this is NOT doing.
 It was my way of providing a way to run 'make check' to run whatever tests are deemed necessary for 'make check'. 
There is no verification of output at all, just a report on the exit status of each test.  Would have to figure out a way to test actual results with expected like you do with some tests in testuite.at.  This is also simpler than my previous implementation.

The one thing I really don't like about it is that you have to relist all the test names in this file, when it would be much better to only deal with actual test names in tests/CMakeLists.txt

Honestly, I only decided to implement  testing as a way to ensure my Windows build was working, but its current form runs well on linux, and hopefully will on Mac as well.  For the time being I think it is a simple way to verify whether the CMake build is working.